Mono V2 Testing

Local9 opened the pull request:
- Fix issue with `Substring` error
- Updated version for NuGet Deployment
- Added 'ref' .dlls from CFX FiveM Client and Server (we don't share the same hard drives)

Currently have an issue where events sent from the client are not showing any activity on the server console, this requires more investigation but currently builds are working so it can be tested.

Client:
![image](https://github.com/manups4e/FxEvents/assets/6077794/67ef2532-41a0-4d11-9c67-c04d3f93a161)

Server:
![image](https://github.com/manups4e/FxEvents/assets/6077794/dbafffe8-c6e0-457f-917e-43a5438269da)

_ignore the MySql.Data error, its not related to the current issue and doesn't stop the resource from working_

Local9 committed:
chore: add mono v2 reference files

Local9 committed:
fix: change binding to all

Local9 committed:
chore: beta build version

Local9 committed:
fix: incorrect use of substring

https://learn.microsoft.com/en-us/dotnet/api/system.string.substring?view=netstandard-2.0#system-string-substring(system-int32-system-int32)

Local9 committed:
chore: update versioning for next nuget update

Local9 commented:
Personally not a fan of the `Substring` use, if there is a better way to get the handle I'd rather do that as we should try to deal with possible changes to the `Remote` class. Maybe @thorium-cfx could provide some insight or take a look, most likely will go "_WTF are you doing?!_" and provide a better solution.

manups4e commented on FxEvents/FxEvents.Server/EventSystem/ServerGateway.cs:
```diff
@@ -40,12 +40,16 @@ public void Push(string pipeline, int source, byte[] buffer)
 
+        private int GetSource(Remote source)
+        {
+            return int.Parse(source.ToString().Substring(7, 1));
```
what if the source has more than 1 char for its source? like 22 or 115?

Local9 replied:
This, is why I do not want to use Substring, the best solution to this would be using `indexOf("(")` maybe

```cs
        private int GetSource(Remote source)
        {
            int startIndex = source.ToString().IndexOf("(");
            int endIndex = source.ToString().LastIndexOf(")");
            int length = endIndex - startIndex + 1;

            if (startIndex < 0)
                throw new Exception("Invalid source, unable to obtain starting index.");

            return int.Parse(source.ToString().Substring(startIndex, length));
        }
```

Local9 replied:
Scrapped this completely after @thorium-cfx comment.

thorium-cfx commented:
> as we should try to deal with possible changes to the Remote class

Remote type will probably get more functionality added, not removed. But you're always free to convert them into a `Player` or `bool` type, those interfaces stay the same. And like my comment above `Dynfunc` already converts these for you.

> most likely will go "WTF are you doing?!" and provide a better solution.

:smirk:

Local9 committed:
chore: put dependancies in the correct folders

Local9 committed:
chore: don't copy references

Local9 committed:
refactor: use func in event handlers

Local9 committed:
refactor: change out Delegate for DynFunc

Have I understood this correctly? The invoke needs to be changed.

```cs
DynFunc dynFunc = Func.Create<T,R>(MyFunc);
dynFunc(remote, arg1, arg2, arg2); // which is the same as:
dynFunc(remote, new object[] { arg1, arg2, arg2 });

Delegate dele = dynFunc;
dele.DynamicInvoke(arg1, arg2, arg3); // calls: dynFunc(arg1, arg2, arg3); which is wrong
dele.DynamicInvoke(remote, new object[] { arg1, arg2, arg3 }); // calls: dynFunc(remote, new object[] { arg1, arg2, arg3 }); which is correct, but slow
```

Local9 committed:
revert: revert call, I know its wrong

Local9 committed:
tweak: its not finished and still doesn't feel right

- Send Remote down from the Event Handlers due to use of DynFunc
- Client doesn't know what a Player is? maybe I'm missing something?
- Client doesn't get information back but the method on the server is invoked, still there is a arguments issue which needs to be addressed.

Local9 committed:
chore: update references

Local9 committed:
tweak: await coroutine before using `IsComplete`

Local9 committed:
tweak: remove a bunch of debug output

Local9 committed:
revert: made some bad choices in the client

Local9 committed:
tweak: more attempts at dealing with not having Remote in client

Local9 committed:
revert: return to using method

Local9 committed:
feat: initial completed testing of mono_v2

Local9 committed:
chore: remove references that were not being used

Local9 commented:
Will install RedM tomorrow and do some RedM testing too.

Local9 committed:
chore: update referenced dlls

Local9 committed:
fix: build errors due to things being moved around

Local9 committed:
tweak: just cast it
